Assets/Scripts/Player/Controllers/BallController.cs

```diff
@@ -14,13 +17,49 @@ private void Awake()
             _stats = GetComponent<PlayerStats>();
         }
 
+        private void Update()
+        {
+            if (Ball != null)
+            {
+                Ball.transform.position = transform.position + new Vector3(BallHoldXOffset, BallHoldYOffset);
```

Comment:
We can cache `new Vector3(BallHoldXOffset, BallHoldYOffset);` in the `Start` function in order to avoid creating a new vector object each frame. 

This probably won't really have any noticeable issues for us here but it's good practice to avoid `new` in performance critical contexts.